libs/sdk-core/src/error.rs

```diff
@@ -0,0 +1,64 @@
+use anyhow::Error;
+use std::fmt;
+
+pub type SdkResult<T, E = NewSdkError> = Result<T, E>;
+
+/// Type of error returned by the SDK
+#[derive(Debug)]
+pub enum NewSdkError {
+    SdkFailedToRequestBackup,
+
+    SdkFailedToStartBackupWatcher,
+
+    SdkFailedToInitPersister,
+    SdkFailedToConnectToGreenlight,
+
+    SdkNoStateSynchronizerFound,
+
+    SdkInitFailedNoNodeApiOrSeedFound,
+
+    SdkCannotRetrievePersistedLastSyncVersion,
+
+    SdkBip32ErrorIndexOutOfBounds,
+    SdkBip32ErrorFailedToDeriveKey,
+
+    SdkFailedToInsertOrUpdatePayments,
+    SdkFailedToInsertLnurlExternalPaymentInfo,
+    SdkFailedToReceivePayment,
+
+    SdkFailedToListPayments,
+
+    SdkFailedToSync,
+
+    // NodeState
+    SdkCannotRetrievePersistedNodeState,
+    SdkNoNodeStateFound,
+
+    // LSPs
+    SdkFailedToRetrieveLsps,
```

Comment:
All these three errors can be unified to PersistenceFailed

Reply:
Done